tracer/src/Datadog.Trace/Debugger/Snapshots/DebuggerSnapshotCreator.cs

```diff
@@ -352,7 +352,8 @@ private void StartLocalsOrArgsIfNeeded(string newParent)
             }
 
             // "locals" should always come after "arguments"
-            if (currentParent == "locals" && newParent == "arguments")
+            if ((currentParent == "locals" && newParent == "arguments") ||
```

Comment:
Why is this change necessary?

Reply:
The top comment `// "locals" should always come after "arguments"` is false. Currently, on method leave (in async and non-async methods) we report the return value in the `EndMethod_StartMarker` routine. `@return` is emplaced within the `locals` , thus in the `return` node of the snapshot the locals arrive before the arguments therefore we should not assume any ordering of arguments/locals relative to one another (I'll also delete this misleading comment).
Does that make sense?